### ISSUE TITLE
replumb/read-eval-call treats multiple forms as a function call

### ISSUE THREAD

**Issue by jaredly** [ScalaConsultants/replumb#129]
`(read-eval-call "2 3" ...)` raises "2 is not a function"; `(read-eval-call "range 10" ...)` calls range w/ the argument 10.
Instead, I expect it to be treated as a `do` block.

**Comment by jaredly** [comment]
hmm digging in, looks like this is probably an issue in core. not sure whether it's "works as intended" though

**Comment by arichiardi** [comment]
Without parens? Mmm... I did not put any implicit do there so the result you see SHOULD be directly what comes out of eval, but there is probably something weird going on.

I would expect the reader not to read a valid object out of "range 10"... I am going to check as well.

**Comment by tomasz-biernacki** [comment]
Same problem in planck
```
cljs.user=> 2 3
2 is not a function (evaluating '(2)(3)')

cljs.user=> range 10
(0 1 2 3 4 5 6 7 8 9)
```

**Comment by tomasz-biernacki** [comment]
The "problem" is that we are passing the whole string to the evaluation so it converts `2 3` to `(2)(3)` and treats it like a function call... on the other side `range 10` yields
```
Evaluating range 10
cljs.core.range(10)
```
so it gets executed without errors.

`lein repl` evaluates each form separately
```
user=> range 1 2 3
#object[clojure.core$range 0xbe06811 "clojure.core$range@be06811"]
1
2
3
```
while the `Quick Start repl` only the first form:
```
cljs.user=> 2 3 range 3
2
```

**Comment by tomasz-biernacki** [comment]
It would be easy to mimic the latter behaviour, we would just need to read the first form... with a special care for tagged literals..

**Comment by arichiardi** [comment]
Maybe at in the other issue, `context` modifies this behavior, we should check in ClojureScript `repl.cljs` and employ the same behavir.

**Comment by jaredly** [comment]
yeah the thing here is `:context :statement` vs `:context :expr`. My belief is that `cljs.js/eval-str` should be changed so that `range 3` becomes `range; 3` instead of `range(3)`

**Comment by arichiardi** [comment]
Like in #131, the thing we can do is first to expose `:context` to keep compatibility and second add a `:jaredly` (you give the name @jaredly :smile: ) if we want to handle the use case. I would close #131 with the first step and then work on a second PR for the second.

@tomasz-biernacki I see you assigned yourself, first step?
As always, feedback is welcome :dancer:

**Comment by mfikes** [comment]
FWIW, Planck now has a fix for this on its master branch, if you'd like to check it out for inspiration https://github.com/mfikes/planck/issues/197

**Comment by jaredly** [comment]
@mfikes looks like a great way to go.
@arichiardi just exposing `:context` would be sufficient. I'll use @mfikes's method for when I want multiple forms & a return value

**Comment by tomasz-biernacki** [comment]
@jaredly I've made few tests with `:context` set to `:statement` and not always it doesn't return the last value, for example:

```
cljs.user=> 2 3
nil
cljs.user=> 2 3 true
nil
cljs.user=> 2 3 :k
nil
cljs.user=> 2 3 '(1 2 3)
(1 2 3)
cljs.user=> 2 3 (+ 1 2 3)
6
cljs.user=> 2 3 (def a 1)
#'cljs.user/a
cljs.user=> a
nil
cljs.user=> (defn f [] 10)
#'cljs.user/f
cljs.user=> (f)
10
```

**Comment by jaredly** [comment]
@tomasz-biernacki yeah I ran into that too. I think it sometimes optimizes out trivial constant expressions

**Comment by arichiardi** [comment]
Closed by #135
